### ISSUE TITLE
XML/JSON Schema implementation and Testing

### ISSUE THREAD

**Issue by aethelwulffe** [CMSgov/qpp-submissions-schema#3]
Starting a tracking issue here to monitor progress and lessons learned while implementing the standards in an actual data-parsing system that generates XML/JSON output (Registry submitter).  This is just a question and summary issue, as any actual projects related to this are in external repos.

Goals:
Create real-world test data examples with full clinic/provider/patient data ->finished output *tested* XML and JSON format with submission steps via API or "upload".

Objectives:
- [ ] Provide public demo (EHR) with sample patient data (real redacted data), that also has a measure calculation and attestation toolset for parsing Registry/Claims/EHR QPP measure sets.  Base demo data should represent one clinic (single group NPI) with 3-5 providers of various levels.
- [ ] Provide output files from the same demo data with calculated results for a set of measures including simple single strata measures, measures with two or more denominators/numerators, and of course the previously nearly unreportable "multi-strata/multi-whatever no-one can figure them out" measures like No. 46 from the 2016 standard.
- [ ] Provide matching XML and JSON output of those same above calculated measures for both individual reporting, reporting as a group, reporting a 90 day period, reporting for 365 day period.

ACTIVITIES:

- Alert the related open-source projects related to this effort.
- Community involvement in creating the sample output files from the above test case.
- Community involvement in submission testing + receiving feedback for testing the submission (immediately) for the end-to-end process on the CMS side.  This is required for full validation of the process.
- Document all lessons learned, identify system weaknesses within the measures standards, submission schemas, and processing of the reports on the receiver side.
- Publish all resulting documentation and the case study by July 20 (complete or otherwise).
-

**Comment by maripuncel** [comment]
Hi @aethelwulffe - this is a great idea and I'd love to help! Do you have access to redacted patient data that we could use for this exercise?

**Comment by aethelwulffe** [comment]
Yes (And thank you).  I have about 200 providers worth of patients (I dunno...a million or so patients) from about 60 clinics from last year.  I am also receiving x12 837p data from 2017 clients, some of whom I am sure will not mind having their data redacted, reviewed, and supplied to the project.  I also have providers (Like Dr. Sam Bowen, formerly Primary and Urgent Care, now Assisted Living Facilities) that would be able to provide clinical-side case studies on preparing for MIPS attestation.  We also have a nice big pack of EHR vendors and independent Medical IT/programmers that are involved with open-source projects at https://librehealth.io, the OpenEMR project, Open VISTA and other places.  They all have MIPS/MACRA related issues on forums and github repositories, and a vast experience base.
  Yeah, I can supply data.
-Art:
suncoastconnection.com
arsponline.org
librehealth.io  https://github.com/LibreHealthIO

**Comment by aethelwulffe** [comment]
What format of patient data would be the most useful to start with?
Take a look at the videos from my companies' product video tutorials from last year.  You will see on-screen reports where the measures were calculated.  I could supply the data out of that system in any report format that would be useful to the developers here, including JSON strings from those completed reports (not your JSON format), copy/paste results from the screen HTML into spreadsheets, DB dumps (and software to look at it in), and a number of other formats.
  I also have a tool that takes billing claim files and turns them into a complete EHR database that works with LibreHealthEHR or OpenEMR (or Phyaura, or Dr. Cloud, or BlueEHR, or any of the hundreds of vendors that are based off the same source as LibreHealthEHR.
  My own QPP reporting tool is proprietary to my company, and is the only way I have of earning money at the moment...and has been developed bootstrap and unpaid for years now.   Despite this, I would not mind actually giving a neutral party access to those repos, and thereby the .sql files that produce the code lists for each measure.  These were painstakingly formatted using the "Standards" PDF's for quality measures, but I feel they are something that the measure standard writers should have supplied.  I also have query logic for each and every measure (against the LibreEHR database) for calculating Initial Patient Population, Denominators, Exclusions, and Numerators for each one.  There has GOT to be some way of leveraging all that work to help other implementors a little bit more of a leg up without completely giving away our work.  If supplying these lists sets a quality standard of documentation for next year, we will all win.

**Comment by aethelwulffe** [comment]
Feel like opening up a /contribs type repo?  I have stuff that could be generally useful...with documentation.
PS:  I note that we still don't actually have a XML/JSON specification document (as of the registry kick-off webinar on the 20th).
